packages/web3-eth-contract/src/utils.ts

```diff
@@ -133,14 +145,19 @@ export const getEstimateGasParams = ({
 			gasPrice: contractOptions.gasPrice,
 			from: contractOptions.from,
 			input: contractOptions.input,
+			data: contractOptions.data,
 		},
 		options as unknown as Record<string, unknown>,
 	) as unknown as TransactionCall;
 
-	txParams = {
-		...txParams,
-		input: encodeMethodABI(abi, params, txParams.input ? toHex(txParams.input) : undefined),
-	};
+	if (!isNullish(txParams.input)) {
+		txParams.input = toHex(txParams.input);
+	}
+	if (!isNullish(txParams.data)) {
+		txParams.data = toHex(txParams.data);
+	}
```

Comment:
Sorry, it is not clear to me. Is this 2 `if`s still needed? If yes, isn't it needed in all other functions, and then we need to move it to `dataInputEncodeMethodHelper`? 
And if this is still needed, why exactly? As with the current code, if the user provides both data and input. The `input` will be assigned with toHex(txParams) and the `data` will be assigned with the encode of its value (you may try the code and then modify).